pkg/types/defaults/installconfig.go

```diff
@@ -44,39 +43,27 @@ func SetInstallConfigDefaults(c *types.InstallConfig) {
 			},
 		}
 	}
-	numberOfMasters := int64(3)
-	numberOfWorkers := int64(3)
+	defaultReplicaCount := int64(3)
 	if c.Platform.Libvirt != nil {
-		numberOfMasters = 1
-		numberOfWorkers = 1
+		defaultReplicaCount = 1
 	}
-	if len(c.Machines) == 0 {
-		c.Machines = []types.MachinePool{
-			{
-				Name:     "master",
-				Replicas: &numberOfMasters,
-			},
+	if c.ControlPlane == nil {
+		c.ControlPlane = &types.MachinePool{
+			Replicas: &defaultReplicaCount,
+		}
+	}
+	c.ControlPlane.Name = "master"
+	if c.Compute == nil {
```

Comment:
Do we want to guard against no-workers until [the router issue][1] gets sorted?  If so, this should be `if len(c.Compute) == 0 {`.

[1]: https://bugzilla.redhat.com/show_bug.cgi?id=1671136#c1

Reply:
Yes, missed this when rebasing with your changes. I will change it.